documentation/design-docs/diagnostics-client-library.md

```diff
@@ -19,46 +19,63 @@ The goal of this library is as following:
 
 ## Sample Code:
 
+Here are some sample code showing the usage of this library.
+
 #### 1. Attaching to a process and dumping out all the runtime GC events in real time to the console
 ```cs
 using Microsoft.Diagnostics.NETCore.Client;
 using Microsoft.Diagnostics.Tracing.Parsers;
 
 public void PrintRuntimeGCEvents(int processId)
 {
-    List<EventPipeProvider> providers = new List<EventPipeProvider>() {
+    var providers = new List<EventPipeProvider>()
+    {
         new EventPipeProvider("Microsoft-Windows-DotNETRuntime",
-            (int)EventLevel.Informational, (long)ClrTraceEventParser.Keywords.GC)
+            EventLevel.Informational, (long)ClrTraceEventParser.Keywords.GC)
     };
 
-    DiagnosticsClient client = new DiagnosticsClient(processId);
-    EventPipeSession session = client.StartEventPipeSession(
-        providers,
-        false, // RequestRundown
-        256 // CircularBufferMB
-    );
-    EventPipeEventSource source = new EventPipeEventSource(session.EventStream);
-
-    source.Dynamic.All += (TraceEvent obj) => {
-        Console.WriteLine(obj.EventName);
-    }
-    try
+    var client = new DiagnosticsClient(processId);
+    using (var session = client.StartEventPipeSession(providers, false))
     {
-        source.Process();
+        var source = new EventPipeEventSource(session.EventStream);
+
+        source.Dynamic.All += (TraceEvent obj) => {
+            Console.WriteLine(obj.EventName);
+        }
+        try
+        {
+            source.Process();
+        }
+        // NOTE: This exception does not currently exist. It is something that needs to be added to TraceEvent.
+        catch (EventStreamException e)
```

Comment:
Do we need to create a new exception or could we just catch the exception that _caused_ the stream reading error?  I don't want to hide the underlying issue by having the `EventSource` simply say "something went wrong".

Reply:
By `EventSource` do you mean `EventPipeEventSource`?

Reply:
whoops!  Yeah, I do.

Reply:
> could we just catch the exception that caused the stream reading error

This part is a little ambiguous. I think of this `EventStreamException` as an exception that gets thrown by EventPipeEventSource when there is an exception while reading the stream because either the stream was shut down by something. There could be other exception, like incorrect or corrupted payload, which can probably be encapsulated by a different exception class. 

>  I don't want to hide the underlying issue 
The sample is meant to capture the intention to add more exception classes to EventPipeEventSource to make it not throw general Exception to prevent exactly this kind of things from happening.